lib/backends/sdc/containers.js

```diff
@@ -3647,6 +3651,14 @@ function _runAttach(opts, callback) {
     if (!serverSocket) {
         serverSocket = net.createConnection({ host: host, port: port });
         serverSocket.on('connect', setupListeners);
+        serverSocket.on('error', function (error) {
+            opts.log.debug('attach for %s threw an error %',
+                cmdString, error.toString());
+
+            serverSocket.end();
```

Comment:
I don't think `serverSocket` is needed here, was it meant to close the other side of the connection (in that case `clientSocket` if I understand correctly)?